### PR TITLE
Tcp random destination IP address

### PR DESCRIPTION
**Pull request by jemcek**

### CODE REVIEW

File: src/function.c [hunk 1]
```diff
@@ -501,6 +504,24 @@ int send_packet(GtkButton *button, gpointer user_data)
 		if (gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(ckbt65)))  params1.inc = params1.inc + 32768;
 		// correct tcp checksum
 		if (gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(ckbt67)))  params1.inc = params1.inc + 65536;
+                //change destination ipv6 address
+		if ((gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(ckbt900))) ) {
+			params1.inc = params1.inc + 131072;
+			//check what user has inserted for mask
+			en900_t = (char *)gtk_entry_get_text(GTK_ENTRY(en900));
+			length = strlen(en900_t);
+			for(m=0; m<length; m++) {
```
Comment (qarkai) [inline src/function.c:513]: `strtol` can handle non-digit values, see https://en.cppreference.com/w/c/string/byte/strtol

File: src/function_send.c [hunk 2]
```diff
@@ -47,6 +47,7 @@
 //#define 128 
 
 extern int number;
+extern unsigned char packet[10000];
```
Comment (qarkai) [inline src/function_send.c:50]: This will again cause build fail with LTO.

File: src/function_send.c [hunk 3]
```diff
@@ -422,6 +434,13 @@ void* sendbuilt (void *parameters)
 				packet[p->ipv4start+14] = (packet[p->ipv4start+14] & ~(0xff>>maskv4[2])) + ((1+(int) (255.0*rand()/(RAND_MAX+1.0))) & (0xff>>maskv4[2]));
 				packet[p->ipv4start+15] = (packet[p->ipv4start+15] & ~(0xff>>maskv4[3])) + ((1+(int) (255.0*rand()/(RAND_MAX+1.0))) & (0xff>>maskv4[3]));
 			}
+			/* change destination IP address */
+			if ( (p->inc & (1<<17)) && (p->ip_proto_in_use == 4)) {
+				packet[p->ipv4start+16] = (packet[p->ipv4start+16] & ~(0xff>>maskv4_2[0])) + ((1+(int) (255.0*rand()/(RAND_MAX+1.0))) & (0xff>>maskv4_2[0]));
+				packet[p->ipv4start+17] = (packet[p->ipv4start+17] & ~(0xff>>maskv4_2[1])) + ((1+(int) (255.0*rand()/(RAND_MAX+1.0))) & (0xff>>maskv4_2[1]));
+				packet[p->ipv4start+18] = (packet[p->ipv4start+18] & ~(0xff>>maskv4_2[2])) + ((1+(int) (255.0*rand()/(RAND_MAX+1.0))) & (0xff>>maskv4_2[2]));
+				packet[p->ipv4start+19] = (packet[p->ipv4start+19] & ~(0xff>>maskv4_2[3])) + ((1+(int) (255.0*rand()/(RAND_MAX+1.0))) & (0xff>>maskv4_2[3]));
+			}
```
Comment (qarkai) [inline src/function_send.c:443]: I suggest to use `for` loop to avoid copy-paste.

File: src/function_send.c (hunk 2 above)
Comment (jemcek) [reply src/function_send.c]: corrected.

**Pull request merged**